common/backoff/retrypolicy.go

```diff
@@ -209,3 +222,63 @@ func (r *retrierImpl) NextBackOff() time.Duration {
 func (r *retrierImpl) getElapsedTime() time.Duration {
 	return r.timeSource.Now().Sub(r.startTime)
 }
+
+var _ RetryPolicy = (*ErrorDependentRetryPolicy)(nil)
+
+func NewErrorDependentRetryPolicy(delayForError func(err error) time.Duration) *ErrorDependentRetryPolicy {
+	return &ErrorDependentRetryPolicy{
+		maximumAttempts: defaultMaximumAttempts,
+		delayForError:   delayForError,
+		jitterPct:       defaultJitterPct,
+	}
+}
+
+func (p *ErrorDependentRetryPolicy) WithMaximumAttempts(maximumAttempts int) *ErrorDependentRetryPolicy {
+	p.maximumAttempts = maximumAttempts
+	return p
+}
+
+func (p *ErrorDependentRetryPolicy) WithJitter(jitterPct float64) *ErrorDependentRetryPolicy {
+	p.jitterPct = jitterPct
+	return p
+}
```

Comment:
This is verbose, but follows the structure of other implementations in this file.